File: web/src/api/download.ts
import { http } from '@/lib/http.ts';

// Download image
export function downloadImage(file?: string, sha256sum?: string) {
  const data = {
    file: file ?? '',
    sha256sum: sha256sum ?? ''
  };
  return http.post('/api/download/image', data);
}

export function cancelDownloadImage() {
  return http.post('/api/download/image/cancel');
}

export function statusImage() {
  return http.get('/api/download/image/status');
}

export function imageEnabled() {
  return http.get('/api/download/image/enabled');
}

File: server/router/download.go
package router

import (
	"NanoKVM-Server/authn"
	"NanoKVM-Server/service/download"
	"github.com/gin-gonic/gin"

	"NanoKVM-Server/middleware"
)

func downloadRouter(r *gin.Engine) {
	service := download.NewService()
	api := r.Group("/api").Use(middleware.CheckToken(), middleware.RequireRole(authn.RoleAdmin))

	api.POST("/download/image", service.DownloadImage)              // download image
	api.POST("/download/image/cancel", service.CancelDownloadImage) // cancel image download
	api.GET("/download/image/status", service.StatusImage)          // download image
	api.GET("/download/image/enabled", service.ImageEnabled)        // download image
	api.POST("/download/file", service.DownloadImageFile)           // download image
}
